### PR TITLE
Allan homework01

### PR DESCRIPTION
**Pull request by allanhorgan**

### CODE REVIEW

File: homework01/allan/vector.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
+class Vector
+{
+public:
+
+	Vector()
+	:Vector(0,0)
```
Comment (david-grs) [inline homework01/allan/vector.h:8]: Good!

File: homework01/allan/vector.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	Vector(int x, int y)
+	:m_x(x),
+	 m_y(y)
+	{
+	}
+
+	bool operator==(const Vector& rhs) const
+	{
+		return m_x == rhs.m_x &&
+			   m_y == rhs.m_y;
```
Comment (david-grs) [inline homework01/allan/vector.h]: Why not on one line? It shouldn't be more than 120 characters? In general one liners are accepted in header file, the rest goes to the cc.

File: homework01/allan/vector.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	Vector& Add(const Vector& rhs)
+	{
+		m_x += rhs.m_x;
+		m_y += rhs.m_y;
+		return *this;
+	}
+
+
+private:
+	int m_x;
```
Comment (david-grs) [inline homework01/allan/vector.h]: Our (Optiver) convention is to use `mX`

File: homework01/allan/vector.h [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	int GetY() const
+	{
+		return m_y;
+	}
+
+	Vector& Add(const Vector& rhs)
+	{
+		m_x += rhs.m_x;
+		m_y += rhs.m_y;
+		return *this;
```
Comment (david-grs) [inline homework01/allan/vector.h]: Same comment as above: this should go to the cc file. The exception is inlined methods (for performance): you need to keep them in the header file, but if you need those, it's better to define them outside the class declaration.

**Review by david-grs — CHANGES_REQUESTED** [review]

File: homework01/allan/vector.h (hunk 4 above)
Comment (allanhorgan) [reply homework01/allan/vector.h]: Thanks David. I've got 2 cc files now. In this case I either must put definitions to the cc file or else inline right? Else I'll get a duplicate at link time?

File: homework01/allan/vector.h (hunk 2 above)
Comment (allanhorgan) [reply homework01/allan/vector.h]: Maybe it's a Java convention but I like to see each test on its own line and the logical operator at the end. I find it more obvious

**Pull request merged**